src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -81,26 +91,44 @@ def backfill_seer_grouping_records(
         return
 
     redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
-    if last_processed_id is None:
-        last_processed_id = int(redis_client.get(make_backfill_redis_key(project_id)) or 0)
-        if last_processed_id == 0 and dry_run:
+    if last_processed_index is None:
+        last_processed_index = int(redis_client.get(make_backfill_redis_key(project_id)) or 0)
+        if last_processed_index == 0 and dry_run:
             logger.info(
                 "backfill_seer_grouping_records.delete_all_seer_records",
                 extra={"project_id": project.id},
             )
             delete_grouping_records(project_id)
 
-    group_id_message_data_batch = (
-        Group.objects.filter(
-            project_id=project.id, id__gt=last_processed_id, type=ErrorGroupType.type_id
+        # Set the metadata of groups where times_seen = 1
+        # Do not set the version number, so we can consider it for future backfills later
+        groups_seen_once = Group.objects.filter(
+            project_id=project_id, type=ErrorGroupType.type_id, times_seen=1
         )
+        for group in groups_seen_once:
+            seer_similarity_seen_once = {"times_seen_once": True}
+            if group.data.get("metadata"):
+                group.data["metadata"]["seer_similarity"] = seer_similarity_seen_once
+            else:
+                group.data["metadata"] = {"seer_similarity": seer_similarity_seen_once}
+
+            if not dry_run:
+                Group.objects.bulk_update(groups_seen_once, ["data"])
+
+    group_id_message_data_batch = (
+        Group.objects.filter(project_id=project.id, type=ErrorGroupType.type_id, times_seen__gt=1)
         .values_list("id", "message", "data")
-        .order_by("id")[:BATCH_SIZE]
+        .order_by("times_seen")
```

Comment:
it any of these are still active groups, this order can potentially change between runs.

Reply:
yeah, we are accepting this and will monitor if this causes problems during LA/EA